tools/CheckSignature.ps1

```diff
@@ -0,0 +1,212 @@
+<#
+    Check (and recurse) current directory                   .\CheckSignature.ps1
+    Find and check directory of Azure PowerShell modules    .\CheckSignature.ps1 -AzurePowerShell
+    Check directory of MSI-installed modules                .\CheckSignature.ps1 -MsiInstall
+    Check directory of gallery-installed modules            .\CheckSignature.ps1 -GalleryInstall
+    Check files in provided path                            .\CheckSignature.ps1 -CustomPath $Path
+#>
+[CmdletBinding(DefaultParameterSetName="CurrentDirectory")]
+Param
+(
+    [Parameter(ParameterSetName="AzurePowerShell", Mandatory=$true)]
+    [switch]$AzurePowerShell,
+    [Parameter(ParameterSetName="MsiInstall", Mandatory=$true)]
+    [switch]$MsiInstall,
+    [Parameter(ParameterSetName="GalleryInstall", Mandatory=$true)]
+    [switch]$GalleryInstall,
+    [Parameter(ParameterSetName="CustomPath", Mandatory=$true)]
+    [string]$CustomPath
+)
+
+function Check-StrongName {
+    [CmdletBinding()]
+    param([Parameter(ValueFromPipeline=$true)][string]$path)
+    $output = & "sn.exe" -vf $path
+    $length = $output.Length - 1
+    if (-not $output[$length].Contains("is valid")) {
+        Write-Output "$path has an invalid strong name."
+    }
+}
+
+function Check-AuthenticodeSignature {
+    [CmdletBinding()]
+    param([Parameter(ValueFromPipeline=$true)][string]$path)
+    $output = Get-AuthenticodeSignature $path
+    if (-not ($output.Status -like "Valid")) {
+        Write-Output "$path has an invalid authenticode signature. Status is $($output.Status)"
+    }
+}
+
+function Check-All {
+    [CmdletBinding()]
+    param([Parameter()][string]$path)
+
+    if (!(Get-Command "sn.exe" -ErrorAction SilentlyContinue))
+    {
+        Write-Error "Unable to find sn.exe; please ensure that the Windows SDK is installed and found in the PATH environment variable."
+        return
+    }
+
+    $invalidList = @()
+
+    $files = Get-ChildItem $path\* -Include *.dll -Recurse | Where-Object { $_.FullName -like "*Azure*" }
```

Comment:
For the future, it would be useful to factor out this file test iterator into its own function, so that you're calling it twice here, rather than duplicating the code.